src/components/views/rooms/RoomHeader.js

```diff
@@ -299,6 +305,14 @@ module.exports = React.createClass({
                 </AccessibleButton>;
         }
 
+        let search_button;
+        if (this.props.onSearchClick && this.props.inRoom) {
```

Comment:
Again I don't think we should be checking for the optional `onSearchClick`, let's allow `AccessibleButton` to decide whether to call `undefined` or not.